Frends.AzureEventHub.Receive/Frends.AzureEventHub.Receive/AzureEventHub.cs

```diff
@@ -0,0 +1,139 @@
+﻿using Azure;
+using Azure.Identity;
+using Azure.Messaging.EventHubs;
+using Azure.Messaging.EventHubs.Consumer;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Models;
+using Frends.AzureEventHub.Receive.Definitions;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Frends.AzureEventHub.Receive;
+
+/// <summary>
+/// Azure EventHub Receive Task.
+/// </summary>
+public static class AzureEventHub
+{
+    /// <summary>
+    /// Receive events from Azure Event Hub.
+    /// [Documentation](https://tasks.frends.com/tasks/frends-tasks/Frends.AzureEventHub.Receive)
+    /// </summary>
+    /// <returns>Object { bool Success, List&lt;dynamic&gt; Data }</returns>
+    public static async Task<Result> Receive([PropertyTab] Consumer consumer, [PropertyTab] Checkpoint checkpoint, [PropertyTab] Options options, CancellationToken cancellationToken)
+    {
+        if (options.MaxEvents.Equals(0) && options.MaximumWaitTimeInMinutes.Equals(0))
+            throw new Exception("Both Options.MaxEvents and Options.MaximumWaitTimeInMinutes cannot be unlimited.");
+
+        var results = new List<dynamic>();
+        var stopProcessing = false;
+        EventProcessorClient processorClient = null;
+        var timeOut = options.MaximumWaitTimeInMinutes > 0 ? DateTime.UtcNow.AddMinutes(options.MaximumWaitTimeInMinutes) : DateTime.UtcNow;
+        var cancellationTokenSource = new CancellationTokenSource();
+
+
+        try
+        {
+            var checkpointStorageClient = CreateBlobContainerClient(checkpoint);
+
+            if (checkpoint.CreateContainer && checkpoint.AuthenticationMethod != AuthenticationMethod.SASToken)
+                await checkpointStorageClient.CreateIfNotExistsAsync(PublicAccessType.None, null, null, cancellationToken);
+
+            processorClient = CreateEventProcessorClient(consumer, checkpointStorageClient);
+
+            processorClient.ProcessEventAsync += async (args) =>
+            {
+                if (options.MaximumWaitTimeInMinutes > 0 && timeOut <= DateTime.UtcNow)
+                    stopProcessing = true;
+                else if (options.MaxEvents > 0 && results.Count >= options.MaxEvents)
+                    stopProcessing = true;
+                else if (args.Data != null && !stopProcessing)
+                {
+                    var result = Encoding.UTF8.GetString(args.Data.Body.ToArray());
+                    results.Add(result);
+                    await args.UpdateCheckpointAsync(cancellationToken);
+                }
+            };
+
+            processorClient.ProcessErrorAsync += async (args) =>
+            {
+                var ex = @$"Partition {args.PartitionId}, Exception: {args.Exception}";
```

Comment:
The @ character is not needed for this.

Reply:
Removed. I'm using both of those because it doesn't affect memory usage or the final result and it's just more user-friendly to not to think about when to use the combo.